drivers/wifi/esp32/src/esp_wifi_drv.c

```diff
@@ -672,3 +674,8 @@ NET_DEVICE_DT_INST_DEFINE(0,
 		&esp32_data, NULL, CONFIG_WIFI_INIT_PRIORITY,
 		&esp32_api, ETHERNET_L2,
 		NET_L2_GET_CTX_TYPE(ETHERNET_L2), NET_ETH_MTU);
+
+#if defined(CONFIG_NET_CONNECTION_MANAGER_CONNECTIVITY_WIFI_MGMT)
+CONN_MGR_CONN_DECLARE_PUBLIC(CONNECTIVITY_WIFI_MGMT);
+CONN_MGR_BIND_CONN(Z_DEVICE_DT_DEV_ID(DT_DRV_INST(0)), CONNECTIVITY_WIFI_MGMT);
+#endif
```

Comment:
I wonder if we could introduce a new macro
```
#define CONN_MGR_BIND_CONN_WIFI(dev) \
	COND_CODE_1(CONFIG_NET_CONNECTION_MANAGER_CONNECTIVITY_WIFI_MGMT, \
		(CONN_MGR_CONN_DECLARE_PUBLIC(CONNECTIVITY_WIFI_MGMT); \
		CONN_MGR_BIND_CONN(dev, CONNECTIVITY_WIFI_MGMT)))
```
And then here we could just do
```
CONN_MGR_BIND_CONN_WIFI(Z_DEVICE_DT_DEV_ID(DT_DRV_INST(0));
```
This way we do not need to repeat the boilerplate code in every driver.

Reply:
Done